packages/server/src/app/db/works/works.service.ts

```diff
@@ -386,12 +402,26 @@ export class WorksService {
      * @param workId The work to approve
      * @param authorId The author of the work
      */
-    async approveWork(workId: string, authorId: string): Promise<void> {
-        //@ts-ignore
-        await this.workModel.updateOne({"_id": workId, "author": authorId}, {"audit.published": models.ApprovalStatus.Approved, "audit.publishedOn": new Date()})
-            .where("audit.isDeleted", false);
+    async approveWork(workId: string, authorId: string): Promise<void> {        
+        const workToApprove = await this.workModel.findOne({
+            _id: workId, 
+            //@ts-ignore
+            author: authorId,
+            'audit.isDeleted': false
+        });           
+        workToApprove.audit.published = models.ApprovalStatus.Approved;
+        workToApprove.audit.publishedOn = new Date();
+        await workToApprove.save();
```

Comment:
This little bit of code could be an example for how we might "fix" the other usages of `update<etc>()`